packages/compass-e2e-tests/tests/collection-aggregations-tab.test.ts

```diff
@@ -688,6 +862,198 @@ describe('Collection aggregations tab', function () {
     );
   });
 
+  it('finds $merge destination as { $merge: <collection> }', async function () {
```

Comment:
Nit: are we sure that we want to test all of the conditions as e2e tests? if we want to cover all of the corner cases and possible variations it may need a good number of them 

how about we extract a `getWriteStageInfo(serverType, ns, pipeline, ...) -> {writeStage, destinationNamespace} | undefined` function, which is a pure function, and we write 200 unit tests for that, covering all sort of corner cases. Here we can just have one e2e test to make sure  the confirmation gets shown on the most simple case.

Reply:
It is not needed since Sergey pointed to existing helpers to extract destinations from stages. Will refactor accordingly and remove redundant tests. Please note that this is a draft PR.

Reply:
Sure, no worries!